internal/pkg/externalplugins/tars/tars.go

```diff
@@ -214,18 +240,67 @@ func HandleAll(log *logrus.Entry, ghc githubClient, config *plugins.Configuratio
 			"pr":   num,
 		})
 
-		pr, err := ghc.GetPullRequest(org, repo, num)
-		if err != nil {
-			l.WithError(err).Error("Error get PR.")
-		}
-		err = handle(l, ghc, pr, externalConfig)
+		err = handleAll(l, ghc, &pr, externalConfig)
 		if err != nil {
 			l.WithError(err).Error("Error handling PR.")
 		}
 	}
 	return nil
 }
 
+func handleAll(log *logrus.Entry, ghc githubClient, pr *pullRequest, cfg *externalplugins.Configuration) error {
+	if pr.Merged {
+		return nil
+	}
+	// Before checking mergeability wait a few seconds to give github a chance to calculate it.
+	// This initial delay prevents us from always wasting the first API token.
+	sleep(time.Second * 5)
+
+	org := string(pr.Repository.Owner.Login)
+	repo := string(pr.Repository.Name)
+	number := int(pr.Number)
+	mergeable := false
+	tars := cfg.TarsFor(org, repo)
+
+	// If the OnlyWhenLabel configuration is set, the pr will only be updated if it has this label.
+	if len(tars.OnlyWhenLabel) != 0 {
+		hasTriggerLabel := false
+		for _, labelName := range pr.Labels.Nodes {
+			if string(labelName.Name) == tars.OnlyWhenLabel {
+				hasTriggerLabel = true
+			}
+		}
+		if !hasTriggerLabel {
+			log.Infof("Ignore PR %s/%s#%d without trigger label %s.", org, repo, number, tars.OnlyWhenLabel)
+			return nil
+		}
+	}
+
+	// Must have last commit.
+	if len(pr.Commits.Nodes) == 0 || len(pr.Commits.Nodes) != 1 {
+		return nil
+	}
```

Comment:
Does the condition mean the pr must have only one commit？

Reply:
No, it mean  we only get the last commit.